packages/wrangler/src/pages/deployment-tails.tsx

```diff
@@ -137,34 +138,54 @@ export async function Handler({
 		PAGES_CONFIG_CACHE_FILENAME
 	);
 	const accountId = await requireAuth(pagesConfig);
+	let deploymentId = deployment;
 
 	projectName = projectName || pagesConfig.project_name;
 
-	if (!deploymentId && !projectName && isInteractive) {
+	// When not specifying a deployment, we need a project name to use the latest
+	// Or when specifying by deployment URL
+	if ((!deployment || isUrl(deployment)) && !projectName && isInteractive) {
 		projectName = await promptSelectProject({ accountId });
 	}
 
-	if (!deploymentId && !projectName) {
-		throw new FatalError("Must specify a project name or deployment ID.", 1);
+	if (!deployment && !projectName) {
+		throw new FatalError("Must specify a project name or deployment.", 1);
 	}
 
-	if (!deploymentId) {
-		logger.log(
-			"No deploymentId specified. Using latest deployment for ",
-			environment,
-			"environment."
-		);
+	const deployments: Array<Deployment> = await fetchResult(
+		`/accounts/${accountId}/pages/projects/${projectName}/deployments`
+	);
 
-		const deployments: Array<Deployment> = await fetchResult(
-			`/accounts/${accountId}/pages/projects/${projectName}/deployments`
-		);
+	const envDeployments = deployments.filter(
+		(d) => d.environment === environment
+	);
 
-		const envDeployments = deployments.filter(
-			(d) => d.environment === environment
+	if (envDeployments.length === 0) {
+		throw new FatalError("No deployments for environment: " + environment, 1);
+	}
+
+	// Deployment is URL
+	if (deployment?.startsWith("https://")) {
+		const { hostname: deploymentHostname } = new URL(deployment);
+		const targetDeployment = envDeployments.find(
+			(d) => new URL(d.url).hostname === deploymentHostname
 		);
 
-		if (envDeployments.length === 0) {
-			throw new FatalError("No deployments for environment: " + environment, 1);
+		if (!targetDeployment) {
```

Comment:
should we maybe do the same if we pass in an invalid/non-existing `deploymentId`? Currently there's no check for that and we simply fallback to the latest deployment. IMO we could perform the check and:

- keep the fallback but warn user the provided id is invalid and that we're falling back to latest
- throw and exit

Not sure what would make more sense 🤔 . Maybe first is less disruptive...

Reply:
In the case of no deploymentID provided, we fallback and warn the user, which I think is what we want. When a user provides an invalid deployment ID, this will fail by way of 404-ing at the `POST /tails` endpoint, thereby throwing and exiting. I think that's fairly ideal, right?

Reply:
Jaykay, I just tried an invalid deployment I borked something. Fixing

Reply:
Okay _now_ it's working as described by my first comment :)